packages/active-campaign-client/src/handlers/resyncUserHandler.ts

```diff
@@ -0,0 +1,69 @@
+import { deleteContact } from '../helpers/deleteContact';
+import { getUserFromCognitoByUsername } from '../helpers/getUserFromCognito';
+import { fetchSubscribedWebinarsFromDynamo } from '../helpers/fetchSubscribedWebinarsFromDynamo';
+import { addContact } from '../helpers/addContact';
+import { APIGatewayProxyResult, SQSEvent } from 'aws-lambda';
+import { addContactToList } from '../helpers/manageListSubscription';
+import { queueEventParser } from '../helpers/queueEventParser';
+import { acClient } from '../clients/activeCampaignClient';
+import { bulkAddContactToList } from '../helpers/bulkAddContactsToLists';
+
+export async function resyncUserHandler(event: {
+  readonly Records: SQSEvent['Records'];
+}): Promise<APIGatewayProxyResult> {
+  try {
+    const queueEvent = queueEventParser(event);
+    const cognitoId = queueEvent.detail.additionalEventData.sub;
+    const deletionResult = await deleteContact(cognitoId);
+    if (deletionResult.statusCode != 200 && deletionResult.statusCode != 404) {
+      // eslint-disable-next-line functional/no-throw-statements
+      throw new Error('Error adding contact');
+    }
+
+    const user = await getUserFromCognitoByUsername(cognitoId);
+
+    if (!user) {
+      console.log(`User: ${cognitoId} not present on Cognito, sync done.`);
+      return {
+        statusCode: 200,
+        body: JSON.stringify({
+          message: 'User not present on Cognito, sync done.',
+        }),
+      };
+    }
+
+    const userWebinarsSubscriptions = await fetchSubscribedWebinarsFromDynamo(
+      cognitoId
+    );
+
+    const webinarIds = JSON.parse(userWebinarsSubscriptions.body)
+      .map(
+        (webinar: { readonly webinarId: { readonly S: string } }) =>
+          webinar?.webinarId?.S
+      )
+      .filter(Boolean);
+
+    console.log('Webinar IDs:', webinarIds); // TODO: Remove after testing
+
+    const webinarListIds = await Promise.all(
+      webinarIds.map(async (webinarId: string) => {
+        const listId = await acClient.getListIdByName(webinarId);
+        return listId;
+      })
```

Comment:
Maybe here we can use a get of all AC Lists instead of a get  foreach list id ?

Reply:
It depends whether we plan to use the AC List for purposes beyond webinars or not. In that case it's safer to fetch only the required ones